### PR TITLE
Fix regl-scatter2d for IE

### PR DESCRIPTION
**Pull request by dy**
This fixes https://github.com/plotly/plotly.js/issues/2802.

### CODE REVIEW

**Comment by etpinard** [comment]
:tada: DIMA IS BACK :tada:

**Comment by dy** [comment]
:bowing_man: sorry for being off for that long. So many juicy things for plotly to accomplish.

**Comment by etpinard** [comment]
no problem @dy !

**Comment by etpinard** [comment]
Just curious @dy , what versions of `npm` are you using? I'm on `v5.8.0`. This might explain why the package-lock diff is a mess.

**Comment by dy** [comment]
Mine is 6.1.0, yes, that may be the culprit. Should we bump it to the latest 6.2.0?

**Comment by etpinard** [comment]
Ok thanks @dy, good to know. Before bumping we would have to check it there's a [CircleCI container](https://hub.docker.com/r/circleci/node/tags/) that runs npm 6.2.0. We should run the same npm version locally and on CI.

As Node 10 ships with npm6 and becomes LTS in [October](https://github.com/nodejs/Release#release-schedule), I'd vote sticking to npm5 until then.

I can re-generate the package-lock, if you working with npm6

**Comment by etpinard** [comment]
The relevant regl-scatter2d commit is -> https://github.com/a-vis/regl-scatter2d/commit/30d3874335cb7617fd061f4950c5eb23f9edb844

File: package-lock.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
       "resolved": "https://registry.npmjs.org/@etpinard/gl-text/-/gl-text-1.1.6.tgz",
       "integrity": "sha512-sN007FwlqdSKJt2/cnGZu3jsAN7G4R/wxk/D6ZivPuQtrwJ42B68iuAqysJPgFepUTAsDRtGAOd1U7tZxfDJwA==",
       "requires": {
-        "color-normalize": "1.1.0",
+        "color-normalize": "1.3.0",
```
Comment (etpinard) [inline package-lock.json]: @dy I just noticed that we're using multiple versions of `color-normalize` in our bundle:

image

Comment (etpinard) [reply package-lock.json]: similarly for `color-rgba`:

image

Comment (etpinard) [reply package-lock.json]: Can `regl-error2d`, `regl-line2d` and `regl-scatter2d` use `color-normalize@1.3.0`?

File: package.json [hunk 2]
```diff
@@ -100,10 +100,10 @@
     "ndarray-ops": "^1.2.2",
     "point-cluster": "^3.1.4",
     "polybooljs": "^1.2.0",
-    "regl": "^1.3.6",
+    "regl": "^1.3.7",
```
Comment (etpinard) [inline package.json]: only this https://github.com/regl-project/regl/pull/496 is new in `1.3.7` which only affects logging styles which should not have any effects for plotly.js

File: package.json [hunk 3]
```diff
@@ -83,9 +84,8 @@
     "gl-scatter3d": "^1.0.11",
     "gl-select-box": "^1.0.2",
     "gl-spikes2d": "^1.0.1",
-    "gl-surface3d": "^1.3.5",
     "gl-streamtube3d": "^1.0.0",
-    "@etpinard/gl-text": "^1.1.6",
```
Comment (etpinard) [inline package.json]: @dy is it ok to use `gl-text` now?

**Comment by etpinard** [comment]
@dy so are all `marker.symbol` values now supported in IE11? 

The [`gl2d_marker_symbols`](https://rreusser.github.io/plotly-mock-viewer/#gl2d_marker_symbols) mock is looking pretty impressive, so I guess the answer is _yes_.

image

but it does spit out

image


in the console on browserstack.

**Comment by dy** [comment]
Yes, I noticed that error. That seems to be regl init error, needs investigation.

**Comment by etpinard** [comment]
Any updates of this @dy . Would be nice to get this in `1.40.0` early next week.

**Comment by etpinard** [comment]
ping @dy - any updates?

**Comment by etpinard** [comment]
@alexcjohnson I'd vote for merging this thing for 1.40.0 even though https://github.com/plotly/plotly.js/pull/2863#issuecomment-409990690 didn't get addressed yet. Marker symbols in IE11 and Edge appear fixed, or at least they're a lot better than before.  https://github.com/plotly/plotly.js/issues/2802 would remain open for the time being.

**Comment by alexcjohnson** [comment]
As long as those console errors don't break other pre-existing functionality then sure, lets merge. 💃

**Comment by etpinard** [comment]
Oops. Looks this PR didn't fix things in Edge:

image

**Pull request merged**